### ISSUE TITLE
select: selected option depends on order of options

### ISSUE THREAD

**Issue by nerdinand** [hanami/helpers#131]
I just discovered this while upgrading to hanami 1.1.
My tests were failing, claiming the expected option was not selected in a `<select>` element. The same test passed in hanami 1.0. I dug around a bit. Apparently the fix in https://github.com/hanami/helpers/commit/c0043de2d65ef0306edfd7af30914653f8e7f4f9 introduced a new bug:

If your options contain a `nil` option, it must be the last option. Otherwise the rendered selection will not be correct if anything but `nil` is the value of your attribute.

Example:

In my view I have:

```haml
- foo = OpenStruct.new(bar: 'a')

= form_for(:foo, '', values: {foo: foo}) do
  - select :bar, [['a', 'a'], ['b', 'b'], ['nil option', nil]]

= form_for(:foo, '', values: {foo: foo}) do
  - select :bar, [['nil option', nil], ['a', 'a'], ['b', 'b']]
```

This leads to the HTML:
```html
<form action="" method="POST" accept-charset="utf-8" id="foo-form">
<input name="_csrf_token" value="0ed6c40b814d0f394d6221746609fc6bcd4eed530a6c773ce39ad63260835b95" type="hidden">
<select name="foo[bar]" id="foo-bar">
<option value="a" selected="selected">a</option>
<option value="b">b</option>
<option value="">nil option</option>
</select>
</form>
<form action="" method="POST" accept-charset="utf-8" id="foo-form">
<input name="_csrf_token" value="0ed6c40b814d0f394d6221746609fc6bcd4eed530a6c773ce39ad63260835b95" type="hidden">
<select name="foo[bar]" id="foo-bar">
<option value="" selected="selected">nil option</option>
<option value="a">a</option>
<option value="b">b</option>
</select>
</form>
```

Which consequently renders as:

![20180323_140640](https://user-images.githubusercontent.com/113440/37830863-dba083bc-2ea3-11e8-9082-85a0e581165f.png)

In this example, both selects should show 'a' as the selected option.

**Comment by GustavoCaso** [comment]
Hi @nerdinand, 

Thanks for opening the issue.

I'm new to helping around here, but I would love to take a look at this one

**Comment by jodosha** [comment]
Fixed by #132
